src/atoms/TextInput/TextInput.native.js

```diff
@@ -0,0 +1,327 @@
+      if (variant === 'outlined') return 'transparent';
+      else if (disabled) return theme.colors.tone[930];
+      else if (isFocused) return theme.colors.tone[940];
+      else return theme.colors.tone[930];
+    },
+  },
+  inputContainer: {
+    width({ size }) {
+      switch (size) {
+        case 'block':
+          return '100%';
+        case 'small':
+          return '160px';
+        case 'medium':
+        default:
+          return '240px';
+      }
+    },
+  },
+};
+
+const InputContainer = styled(View)`
+  background-color: transparent;
+  width: ${styles.inputContainer.width};
+`;
+
+const StyledInput = styled(NativeTextInput)`
+  height: 40px;
+  padding: ${styles.textInput.padding};
+  font-size: ${styles.textInput.fontSize};
+  line-height: ${styles.textInput.lineHeight};
+  font-family: ${styles.textInput.fontFamily};
+  color: ${styles.textInput.color};
+`;
+
+const FillContainer = styled(View)`
+  background-color: ${styles.fillContainer.backgroundColor};
+  border-top-left-radius: 2px;
+  border-top-right-radius: 2px;
+`;
+
+const getAccessoryConfig = ({ errorText, prefix, suffix, iconLeft, iconRight }) => {
+  const hasError = !isEmpty(errorText);
+  const hasPrefix = !isEmpty(prefix);
+  const hasSuffix = !isEmpty(suffix);
+  const hasLeftIcon = !hasPrefix && !isEmpty(iconLeft);
+  const hasRightIcon = !hasSuffix && !isEmpty(iconRight);
+  return { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon };
+};
+
+const TextInput = ({
+  placeholder,
+  onChangeText,
+  helpText,
+  errorText,
+  variant,
+  prefix,
+  suffix,
+  disabled,
+  children,
+  iconLeft,
+  iconRight,
+  maxLength,
+  showCharacterCount,
+  label,
+  testID,
+  labelPosition,
+  size,
+}) => {
+  const theme = useContext(ThemeContext);
+  const [isFocused, setFocused] = useState(false);
+  const [input, setInput] = useState(children || '');
+  // Used for storing layout value of TextInput
+  const [layoutDimensions, setLayoutDimensions] = useState(null);
+  // Used to hide placeholder while label is inside the TextInput
+  const [isPlaceholderVisible, setIsPlaceholderVisible] = useState(isFocused);
+
+  const hasLeftLabel = labelPosition === 'left' && variant === 'filled';
+
+  const placeholderTextColor = disabled ? theme.colors.shade[930] : theme.colors.shade[940];
+
+  // Derive accessory conditions based on props
+  const { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon } = getAccessoryConfig({
+    errorText,
+    prefix,
+    suffix,
+    iconLeft,
+    iconRight,
+  });
+
+  const hasText = !!(input && input.length > 0);
+  const onFocus = useCallback(() => {
+    setFocused(true);
+
+    /* Wait for 90ms to show the placeholder since it takes 100ms for Label to animate from inside to top of the TextInput.
+       Otherwise they both overlap */
+    /* Don't have any delay if label is on left of TextInput */
+
+    setTimeout(() => {
+      setIsPlaceholderVisible(true);
+    }, 90);
+  }, []);
```

Comment:
`[setFocused, setIsPlaceholderVisible]`

Reply:
```setFocused``` & ```setIsPlaceholderVisible``` are functions that will not change over time means we don't need to pass them as dependencies to ```useCallback```. I don't think we should be passing these functions here

Reply:
https://reactjs.org/docs/hooks-reference.html#usecallback

> `Pass an inline callback and an array of dependencies. useCallback will return a memoized version of the callback that only changes if one of the dependencies has changed.`